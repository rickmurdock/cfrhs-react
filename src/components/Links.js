import React, { Component } from 'react';
import linksHistorical from '../data/links/linksHistorical';
import linksRailroad from '../data/links/linksRailroad';
import linksTourist from '../data/links/linksTourist';
import LinkCard from './LinkCard';

class Links extends Component {
  render() {
    let historicalLinks = linksHistorical.map((historical) => {
      return (
        <li><a key={historical.id} href={historical.url} target="_blank">{historical.name}</a></li>
      )
    });

    let railroadLinks = linksRailroad.map((railroad) => {
      return (
        <li><a key={railroad.id} href={railroad.url} target="_blank">{railroad.name}</a></li>
      )
    });

    let touristLinks = linksTourist.map((tourist) => {
      return (
        <li><a key={tourist.id} href={tourist.url} target="_blank">{tourist.name}</a></li>
      )
    });
    return (
      <div className="container-fluid">
        <h2 className="text-center page-title my-3">Railroad Related Links</h2>
        <div className="row">
          <div className="col-md-3"></div>
          <div id="accordion" role="tablist" aria-multiselectable="true" className="col-md-6">
            <LinkCard title={"Historical Organizations & Museums"} card={"One"} show={"show"} expanded="true"> 
              {historicalLinks}
            </LinkCard>
            <LinkCard title={"Railroad Operations & Rail Fan Opportunities"} card={"Two"} show={""} expanded="false"> 
              {railroadLinks}
            </LinkCard>
            <LinkCard title={"Tourist Railroads, Trolleys & Live Steamer Operations in Florida"} card={"Three"} show={""} expanded="false"> 
              {touristLinks}
            </LinkCard>
          </div>
        </div>
      </div>
    );
  }
}

export default Links;