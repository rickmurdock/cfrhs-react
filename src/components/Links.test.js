import React from 'react';
import renderer from 'react-test-renderer';
import Links from './Links';

test('Links renders correctly', () => {
    const tree = renderer.create(
        <Links />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});