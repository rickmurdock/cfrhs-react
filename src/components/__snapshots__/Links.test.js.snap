// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links renders correctly 1`] = `
<div
  className="col-md-12 links-page"
>
  <div
    className="container-fluid"
  >
    <h2
      className="text-center page-title page-title-links my-3"
    >
      Railroad Related Links
    </h2>
    <div
      className="row links-row"
    >
      <div
        className="col-md-3"
      />
      <div
        aria-multiselectable="true"
        className="col-md-6"
        id="accordion"
        role="tablist"
      >
        <div
          className="card"
        >
          <div
            className="card-header"
            id="headingOne"
            role="tab"
          >
            <h5
              className="mb-0"
            >
              <a
                aria-controls="collapseOne"
                aria-expanded="true"
                data-parent="#accordion"
                data-toggle="collapse"
                href="#collapseOne"
              >
                Historical Organizations & Museums
              </a>
            </h5>
          </div>
          <div
            aria-labelledby="headingOne"
            className="collapse show"
            id="collapseOne"
            role="tabpanel"
          >
            <div
              className="card-block"
            >
              <ul
                className="ul-links"
              >
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.aclsal.org/"
                    target="_blank"
                  >
                    ACL & SAL Historical Society
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://sites.google.com/site/archerhistoricalsociety/"
                    target="_blank"
                  >
                    Archer Historical Society Railroad Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://railsofpalatka.org/"
                    target="_blank"
                  >
                    David Browning Railroad Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.floridaflywheelers.org/"
                    target="_blank"
                  >
                    Florida Flywheelers
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.frrm.org/"
                    target="_blank"
                  >
                    Florida Railroad Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://gcrm.org/"
                    target="_blank"
                  >
                    Gold Coast Railroad Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://flaglermuseum.us/"
                    target="_blank"
                  >
                    Henry Morrison Flagler Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.plantmuseum.com/"
                    target="_blank"
                  >
                    Henry B. Plant Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.atrrm.org/"
                    target="_blank"
                  >
                    Heritage Rail Alliance
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.flarr.com/jtc.htm"
                    target="_blank"
                  >
                    Jacksonville Union Station
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.naplestrainmuseum.org/"
                    target="_blank"
                  >
                    Naples Train Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.nrhs.com/"
                    target="_blank"
                  >
                    National Railway Historical Society
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.plantcitygov.com/939/Union-Station-Depot---Train-Viewing-Plat"
                    target="_blank"
                  >
                    Plant City Florida Union Station Depot and Train Viewing Platform
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://rlhs.org/"
                    target="_blank"
                  >
                    Railway & Locomotive Historical Society
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://rrmsf.org/"
                    target="_blank"
                  >
                    Railroad Museum of South Florida
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.willafordrailroadmuseum.com/"
                    target="_blank"
                  >
                    Robert W. Willaford Railroad Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.sfrm.org/"
                    target="_blank"
                  >
                    South Florida Railway Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.tampaunionstation.com/"
                    target="_blank"
                  >
                    Tampa Union Station
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.taplines.net/"
                    target="_blank"
                  >
                    Taplines
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.brooksvillehistoricaltraindepot.com/"
                    target="_blank"
                  >
                    Train Depot Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.wfrm.org/"
                    target="_blank"
                  >
                    West Florida Railroad Museum
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.wghf.org/"
                    target="_blank"
                  >
                    Winter Garden Heritage Foundation
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card-header"
            id="headingTwo"
            role="tab"
          >
            <h5
              className="mb-0"
            >
              <a
                aria-controls="collapseTwo"
                aria-expanded="false"
                data-parent="#accordion"
                data-toggle="collapse"
                href="#collapseTwo"
              >
                Railroad Operations & Rail Fan Opportunities
              </a>
            </h5>
          </div>
          <div
            aria-labelledby="headingTwo"
            className="collapse "
            id="collapseTwo"
            role="tabpanel"
          >
            <div
              className="card-block"
            >
              <ul
                className="ul-links"
              >
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.gwrr.com/railroads/north_america/alabama_gulf_coast_railway"
                    target="_blank"
                  >
                    Alabama & Gulf Coast Railway
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.amtrak.com/home"
                    target="_blank"
                  >
                    Amtrak
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.gwrr.com/railroads/north_america/an_railway"
                    target="_blank"
                  >
                    Apalachicola Northern Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.gwrr.com/railroads/north_america/bay_line_railroad"
                    target="_blank"
                  >
                    Atlanta & St. Andrews Bay Railway
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://gobrightline.com/"
                    target="_blank"
                  >
                    Brightline
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.csx.com/"
                    target="_blank"
                  >
                    CSX
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.gwrr.com/railroads/north_america/first_coast_railroad"
                    target="_blank"
                  >
                    First Coast Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.pinsly.com/"
                    target="_blank"
                  >
                    Florida Central Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.fecrwy.com/"
                    target="_blank"
                  >
                    Florida East Coast Railway
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.pinsly.com/"
                    target="_blank"
                  >
                    Florida Midland Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.pinsly.com/"
                    target="_blank"
                  >
                    Florida Northern Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.folkston.com/trains/trains.htm"
                    target="_blank"
                  >
                    Folkston Funnel
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://omnitrax.com/our-company/our-railroads/georgia-florida-railway-llc/"
                    target="_blank"
                  >
                    Georgia & Florida Railway
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.watcocompanies.com/services/rail/jacksonville-port-terminal-railroad-jxpt/"
                    target="_blank"
                  >
                    Jacksonville Port Terminal Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.nscorp.com/"
                    target="_blank"
                  >
                    Norfolk Southern
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://semgulf.com/"
                    target="_blank"
                  >
                    Seminole Gulf Railway
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.ussugar.com/railroad/"
                    target="_blank"
                  >
                    South Central Florida Express
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.stmarysrailroad.com/"
                    target="_blank"
                  >
                    St. Marys Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.sunrail.com/"
                    target="_blank"
                  >
                    SunRail
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.tri-rail.com/"
                    target="_blank"
                  >
                    Tri-Rail
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.gwrr.com/railroads/north_america/valdosta_railway"
                    target="_blank"
                  >
                    Valdosta Railway
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card-header"
            id="headingThree"
            role="tab"
          >
            <h5
              className="mb-0"
            >
              <a
                aria-controls="collapseThree"
                aria-expanded="false"
                data-parent="#accordion"
                data-toggle="collapse"
                href="#collapseThree"
              >
                Tourist Railroads, Trolleys & Live Steamer Operations in Florida
              </a>
            </h5>
          </div>
          <div
            aria-labelledby="headingThree"
            className="collapse "
            id="collapseThree"
            role="tabpanel"
          >
            <div
              className="card-block"
            >
              <ul
                className="ul-links"
              >
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.bigbootsrr.com/"
                    target="_blank"
                  >
                    Big Boots & Western Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://cpgrr.org/"
                    target="_blank"
                  >
                    Central Pasco & Gulf Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.floridalivesteamers.com/"
                    target="_blank"
                  >
                    Florida Live Steamers
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.lcrailroad.com/"
                    target="_blank"
                  >
                    Largo Central Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://www.facebook.com/pages/Manatee-Central-Railroad/400459156671215"
                    target="_blank"
                  >
                    Manatee Central Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.neflsri.com/"
                    target="_blank"
                  >
                    Northeast Florida Live Steamers & Railroaders, Inc.
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://orlandoandnorthwesternrailway.com/"
                    target="_blank"
                  >
                    Orlando & Northwestern Railway
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.ridgelivesteamers.org/"
                    target="_blank"
                  >
                    Ridge Live Steamers
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.tecolinestreetcar.org/"
                    target="_blank"
                  >
                    Tampa Electric Company Trolley
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="http://www.livesteamers.org/"
                    target="_blank"
                  >
                    Tradewinds & Atlantic Railroad
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://disneyworld.disney.go.com/guest-services/monorail-transportation/"
                    target="_blank"
                  >
                    Walt Disney World Monorail
                  </a>
                </li>
                <li
                  className="links-li"
                >
                  <a
                    href="https://disneyworld.disney.go.com/attractions/magic-kingdom/walt-disney-world-railroad-fantasyland/"
                    target="_blank"
                  >
                    Walt Disney World Railroad
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
